client/a8c-for-agencies/sections/marketplace/products-overview-v2/product-listing/index.tsx

```diff
@@ -0,0 +1,330 @@
+import type { ShoppingCartItem } from '../../types';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+
+interface ProductListingProps {
+	selectedSite?: SiteDetails | null;
+	suggestedProduct?: string;
+	productBrand: string;
+	productSearchQuery?: string;
+	isReferralMode: boolean;
+	selectedBundleSize: number;
+	selectedFilters: SelectedFilters;
+}
+
+export default function ProductListing( {
+	selectedSite,
+	suggestedProduct,
+	productSearchQuery,
+	isReferralMode,
+	selectedBundleSize,
+	selectedFilters,
+}: ProductListingProps ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const { selectedCartItems, setSelectedCartItems } = useContext( ShoppingCartContext );
+
+	const quantity = useMemo(
+		() => ( isReferralMode ? 1 : selectedBundleSize ),
+		[ isReferralMode, selectedBundleSize ]
+	);
+
+	const {
+		filteredProductsAndBundles,
+		isLoadingProducts,
+		jetpackPlans,
+		jetpackBackupAddons,
+		jetpackProducts,
+		wooExtensions,
+		data,
+		suggestedProductSlugs,
+	} = useProductAndPlans( {
+		selectedSite,
+		selectedBundleSize: quantity,
+		selectedProductFilters: selectedFilters,
+		productSearchQuery,
+	} );
+
+	const isEmptyList = ! filteredProductsAndBundles.length;
+
+	// Create a ref for `filteredProductsAndBundles` to prevent unnecessary re-renders caused by the `useEffect` hook.
+	const filteredProductsAndBundlesRef = useRef( filteredProductsAndBundles );
+
+	// Update the ref whenever `filteredProductsAndBundles` changes.
+	useEffect( () => {
+		filteredProductsAndBundlesRef.current = filteredProductsAndBundles;
+	}, [ filteredProductsAndBundles ] );
+
+	const preSelectProducts = useCallback( () => {
+		const productsQueryArg = getQueryArg( window.location.href, 'products' )?.toString?.();
+		const parsedItems = parseQueryStringProducts( productsQueryArg );
+		const availableSizes = getSupportedBundleSizes( data );
+
+		const allProductsAndBundles = parsedItems?.length
+			? ( parsedItems
+					.map( ( item ) => {
+						// Add products & bundles that are supported
+						const product = filteredProductsAndBundlesRef.current.find(
+							( product ) => product.slug === item.slug
+						);
+						const quantity = availableSizes.find( ( size ) => size === item.quantity );
+						if ( product && quantity ) {
+							return {
+								...product,
+								quantity,
+							};
+						}
+						return null;
+					} )
+					.filter( Boolean ) as ShoppingCartItem[] )
+			: null;
+
+		if ( allProductsAndBundles ) {
+			setSelectedCartItems( allProductsAndBundles );
+		}
+	}, [ setSelectedCartItems, data ] );
+
+	useEffect( () => {
+		if ( isLoadingProducts ) {
+			return;
+		}
+		preSelectProducts();
+	}, [ isLoadingProducts, preSelectProducts ] );
+
+	const incompatibleProducts = useMemo(
+		() =>
+			// Only check for incompatible products if we have a selected site.
```

Comment:
Oh, interesting. I didn't know there was a concept of a selected site while browsing products.

Reply:
This has been inherited from Jetpack Manage's logic, and it seems to be utilized when initiating a purchase flow from the Sites dashboard.

I will need to check if this is still accurate, as it's possible we no longer use this flow.